src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesCcm.Android.cs

```diff
@@ -141,13 +143,20 @@ private void ImportKey(ReadOnlySpan<byte> key)
 
                 plaintextBytesWritten += bytesWritten;
 
-                if (!Interop.Crypto.EvpCipherFinalEx(
+                if (!Interop.Crypto.EvpAeadCipherFinalEx(
                     ctx,
                     plaintext.Slice(plaintextBytesWritten),
-                    out bytesWritten))
+                    out bytesWritten,
+                    out bool authTagMismatch))
                 {
                     CryptographicOperations.ZeroMemory(plaintext);
-                    throw new CryptographicException(SR.Cryptography_AuthTagMismatch);
+
+                    if (authTagMismatch)
+                    {
+                        throw new AuthenticationTagMismatchException();
+                    }
+
+                    throw new CryptographicException(SR.Arg_CryptographyException);
```

Comment:
It surprised me that this wasn't just `throw new CryptographicException()`.  But this is what the line looks like after the JIT inlines the default ctor, so :shrug:.